src/models/cost_function_timeseries.jl

```diff
@@ -0,0 +1,112 @@
+function get_variable_cost(
+    ts::IS.TimeSeriesData,
+    ::Union{MarketBidCost, ReserveDemandCurve},
+)
+    data = TimeSeries.TimeArray(TimeSeries.timestamp(ts), map(VariableCost, ts.data))
```

Comment:
This looks non-functional to me.
1. TimeSeries.timestamp(ts) is not valid.
2. Not all subtypes of TimeSeriesData have the field `data`.
3. The scaling_factor_multiplier is not accounted for.  Even if these time series are never scaling factors I think we should strive to use the same interface.
4. Is this intended to return all forecast windows or just one?

  I _think_ you want this instead:
```
data = map(VariableCost, get_time_series_array(component, ts, start_time))
```
component and start_time are not passed in.  You could pick the first initial time.